src/java.base/share/classes/java/text/MessageFormat.java

```diff
@@ -1651,6 +1820,72 @@ private static void copyAndFixQuotes(String source, int start, int end,
         }
     }
 
+    // Corresponding to the FormatType pattern
+    private enum FormatType {
+        NUMBER,
+        DATE,
+        TIME,
+        DTF_DATE,
+        DTF_TIME,
+        DTF_DATETIME,
+        CHOICE,
+        LIST,
+
+        // Pre-defined DateTimeFormatter types
+        BASIC_ISO_DATE,
+        ISO_LOCAL_DATE,
+        ISO_OFFSET_DATE ,
+        ISO_DATE,
+        ISO_LOCAL_TIME,
+        ISO_OFFSET_TIME,
+        ISO_TIME,
+        ISO_LOCAL_DATE_TIME,
+        ISO_OFFSET_DATE_TIME,
+        ISO_ZONED_DATE_TIME,
+        ISO_DATE_TIME,
+        ISO_ORDINAL_DATE,
+        ISO_WEEK_DATE,
+        ISO_INSTANT,
+        RFC_1123_DATE_TIME;
+    }
+
+    // Corresponding to the FormatStyle pattern
+    private enum FormatStyle {
+        DEFAULT(""),
+        SHORT("short"),
+        MEDIUM("medium"),
+        LONG("long"),
+        FULL("full"),
+        INTEGER("integer"),
+        CURRENCY("currency"),
+        PERCENT("percent"),
+        COMPACT_SHORT("compact_short"),
+        COMPACT_LONG("compact_long"),
+        OR("or"),
+        UNIT("unit"),
+        SUBFORMATPATTERN(null);
+
+        private final String text;
+
+        // Differs from FormatType in that the text String is
+        // not guaranteed to match the Enum name, thus a text field is used
+        FormatStyle(String text){
+            this.text = text;
+        }
+
+        // This method returns a FormatStyle that matches the passed String.
+        // If no FormatStyle is found, IllegalArgumentException is thrown
+        private static FormatStyle fromString(String text) {
+            for (FormatStyle style : values()) {
+                // Also check trimmed case-insensitive for historical reasons
+                if (text.trim().toLowerCase(Locale.ROOT).equals(style.text)) {
```

Comment:
`String.compareIgnoreCase(....)` might be clearer and not need to allocate for a converted string.